### PR TITLE
added integration test for Presto

### PR DESCRIPTION
**Pull request by DariuszAniszewski**
@karth295 PTAL

### CODE REVIEW

File: presto/test_presto.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        ("STANDARD", "1.1", ["m"], 1, 2),
+        ("HA", "1.1", ["m-0"], 1, 2),
+        ("SINGLE", "1.2", ["m"], 1, 0),
+        ("STANDARD", "1.2", ["m"], 1, 2),
+        ("HA", "1.2", ["m-0"], 1, 2),
+        ("SINGLE", "1.3", ["m"], 1, 0),
+        ("STANDARD", "1.3", ["m"], 1, 2),
+        ("HA", "1.3", ["m-0"], 1, 2),
+    ], testcase_func_name=DataprocTestCase.generate_verbose_test_name)
+    def test_pesto(self, configuration, dataproc_version, machine_suffixes, coordinators, workers):
```
Comment (karth295) [inline presto/test_presto.py]: Nit: presto (typo)

File: presto/test_presto.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    def setUp(self):
+        pass
+
+    def verify_instance(self, name, coordinators, workers):
+        schema = "test_{}".format(random.randint(0, 1000))
+        table = "table{}".format(random.randint(0, 1000))
+
+        self.__create_schema(name, schema)
+        self.__create_table(name, table)
+        self.__verify_schema(name, schema)
```
Comment (karth295) [inline presto/test_presto.py]: Please add a small presto query -- it's important to make sure that scheduling tasks and shuffle both work. Something like word count: `SELECT number, count(1) AS total FROM test_5 GROUP BY number ORDER BY total DESC LIMIT 10`.

I think that means you'll need to add some numbers into the hive table.

File: presto/test_presto.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        self.assertIn(schema, str(stdout), "Schema {} not found in {}".format(schema, stdout))
+
+    def __verify_coordinators_count(self, coordinators, name):
+        ret_code, stdout, stderr = self.run_command(
+            "gcloud compute ssh {} -- \"presto --execute 'select count(*) from system.runtime.nodes where coordinator=true'\"".format(
+                name
+            )
+        )
+        self.assertEqual(ret_code, 0, "Failed to check number of coodrinators. Error: {}".format(stderr))
+        self.assertIn(str(coordinators), str(stdout), "Bad number of coordinators. Expected: {}\tFound: {}".format(
```
Comment (karth295) [inline presto/test_presto.py]: assertIn makes me nervous -- what if you're searching for `1` but the output is `10`?

To make the output more easily parsable, considering using `--output-format` to dump as csv. I think all of these assertions should be exact.

**Comment by DariuszAniszewski** [comment]
@functicons can you please take another look?

**Comment by functicons** [comment]
I pulled the PR and tested locally, but it failed:

```
$ python -m unittest presto.test_presto
FFFFFFFFFFFF
======================================================================
FAIL: test_presto.mode: HA.version: 1_0 (presto.test_presto.PrestoTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/google/home/dagang/venv/venv3.5/lib/python3.5/site-packages/parameterized/parameterized.py", line 392, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/usr/local/google/home/dagang/repo/github/GoogleCloudPlatform/dataproc-initialization-actions/presto/test_presto.py", line 135, in test_presto
    workers
  File "/usr/local/google/home/dagang/repo/github/GoogleCloudPlatform/dataproc-initialization-actions/presto/test_presto.py", line 17, in verify_instance
    self.__verify_coordinators_count(name, coordinators)
  File "/usr/local/google/home/dagang/repo/github/GoogleCloudPlatform/dataproc-initialization-actions/presto/test_presto.py", line 94, in __verify_coordinators_count
    self.assertEqual(ret_code, 0, "Failed to check number of coordinators. Error: {}".format(stderr))
AssertionError: 1 != 0 : Failed to check number of coordinators. Error: ERROR: (gcloud.compute.ssh) Could not fetch resource:
 - The resource 'projects/google.com:dagang-devconsole/zones/us-west1-a/instances/1' was not found
```

**Comment by DariuszAniszewski** [comment]
@functicons thanks for info. It was silly mistake and I feel a little embarrassed... But it's fixed ;)

Please test it again, but keep in mind that this PR changes init-action also, so either upload it to your GCS or simply use this one: `gs://dariusz-init-actions/presto/presto.sh`

**Comment by functicons** [comment]
@DariuszAniszewski Do you mean I need to first upload the presto.sh then run the test? If so, I would suggest extract init action change out, I'll review, approve and upload it first. `python3.5 -m unittest presto.test_presto` should just work.

**Comment by DariuszAniszewski** [comment]
@functicons I reverted the init-action change from this PR and created #414. Once it's merge, this test will fully work.

**Pull request merged**